### DISCUSSION TITLE
pydantic response model in orm mode, struggle with list of tuples of models

### DISCUSSION THREAD

**Discussion by bluebrown** [fastapi/fastapi#4255] [Questions] [answered]
Hi, I was following this guide along, but something is strange about it. https://fastapi.tiangolo.com/tutorial/sql-databases/#use-pydantics-orm_mode.

When doing a select against the DB, the result is of type `List[Tuple[SqlAlchemyModel]]` .

```python
data = session.execute(select(MyModel)).all()
# [(MyModel,), ...]
```

so simply using a pydantic model in ORM mode didn't work. I had to map the list to strip the tuple away.

```python
data = [item[0] for item in session.execute(select(MyModel)).all()]
# [MyModel, ...]
```

Only then I could use `response_model=List[PydanticModel]` otherwise its fields would be all null.

Am I overlooking something here? I want to return the result directly from the handler without having to remap it.

**Comment by bluebrown** [comment, marked as answer]
I found out that it works with scalars in this case.

```python
data = session.scalars(select(MyModel)).all()
```

**Reply by carlfarrington** [reply]
also, query() seems to work.
select() works when using SQLAlchemy 1.4, but not 2.0
